src/ts/components/transaction/Fee.tsx

```diff
@@ -40,14 +62,26 @@ class Fee extends React.Component<IFeeProps, IFeeState> {
   }
 
   private doUpdate = () => {
-    this.api.query.balances.transferFee(currentFee => {
-      console.log('currentFee', currentFee)
-      const formattedFee = formatBalance(currentFee)
+    const address = this.props.address
+    const toAddress = this.props.toAddress
+    Promise.all([
+      this.api.derive.balances.fees() as unknown as DerivedFees,
+      this.api.query.system.accountNonce(address) as unknown as BN,
+      this.api.tx.balances.transfer(address, 1) as unknown as IExtrinsic,
+      this.api.query.balances.freeBalance(toAddress) as unknown as Balance,
+      this.api.query.balances.reservedBalance(toAddress) as unknown as Balance
+    ]).then(([fees, nonce, ext, freeBalance, rsvdBalance]) => {
+      const extLength = calcSignatureLength(ext, nonce)
+      const baseFee = new BN(fees.transactionBaseFee)
```

Comment:
Is `this.api.derive.balances.fees()` static for all transactions ? if yes, we probably can save it to store so that we don't need to fetch its values for every fee update. e.g `fees.transferFee`, `fees.transactionByteFee`, `fees.transactionBaseFee` , `fees.creationFee` etc

Reply:
Good question. I think it is but I would need to ask that on Polkadot Watercooler.

Reply:
I asked it on riot and will add fees to store if it is static.